tests/test_buffered.py

```diff
@@ -77,3 +77,28 @@ async def test_writable_file_async():
     resp = await obs.get_async(store, path)
     retour = await resp.bytes_async()
     assert retour == line * 50
+
+
+def test_read_past_eof_sync():
+    store = MemoryStore()
```

Comment:
Can you also compare this with `BytesIO` so that we know we're matching stdlib behavior? Like
```
buf = BytesIO(data)
expected = buf.read(20)
```

Reply:
done!